backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -100,6 +100,21 @@ public User registerAlpacaApiKey(Long id, String key) {
         return userToUpdate;
     }
 
+    /**
+     * A method to verify given credentials against existing user records
+     * @param credentials A User object with email and unencrypted password
+     * @return The full User object from the database if login is successful, null otherwise
+     */
+    @Override
+    public User login(User credentials) {
+        User user = userDao.findUserByEmail(credentials.getEmail());
+        if(user != null && compareDigests(credentials.getPassword(), user.getPassword())){
+            return user;
+        }else{
+            return null;
```

Comment:
I think it is better if we throw the exception here and not in the controller. WDYT? (Refactor?)

Reply:
separation of concerns, I agree.